domain-models-runtime/src/test/java/org/folio/rest/persist/PgUtilIT.java

```diff
@@ -555,6 +589,200 @@ public void responseWithExceptionInFailResponseMethod(TestContext testContext) t
     testContext.assertTrue(future.failed());
     testContext.assertEquals("some runtime exception", future.cause().getCause().getMessage());
   }
+  @Test
+  public void canGetWithOptimizedSql(TestContext testContext) {
+	int optimizdSQLSize = 10000;
+    int n = optimizdSQLSize / 2;
+    PostgresClient pg = PostgresClient.getInstance(vertx, "testtenant");
+
+    // "b foo" records are before the getOptimizedSqlSize() limit
+    // "d foo" records are after the getOptimizedSqlSize() limit
+    insert(testContext, pg, "a", n);
+    insert(testContext, pg, "b foo", 5);
+    insert(testContext, pg, "c", n);
+    insert(testContext, pg, "d foo", 5);
+    insert(testContext, pg, "e", n);
+    String columnName = "name";
+    // limit=9
+    //prepare optimized sql
+    String optimizedSQL = "";
+    PreparedCQL pCQL = null;
+    CQLWrapper wrapper = null;
+    try {
```

Comment:
Why can't you replace this try-catch-catch-executeAndNotify section by searchForData?